### ISSUE TITLE
`rollup:options` in hooks throw error when `options.plugins = [ options.plugins ]`

### ISSUE THREAD

**Issue by s3xysteak** [unjs/unbuild#378]
### Environment

node: `v20.9.0`
pnpm: `8.15.4`
`"typescript": "^5.4.2",`

### Reproduction

github [repo](https://github.com/s3xysteak/unbuildErrTest)
`pnpm i`
`pnpm build`
It will throw error.

### Describe the bug

See `build.config.ts` in the mini-repo above.
The option of defineBuildConfig will throw error:
```ts
export default defineBuildConfig({
  // ...
  hooks: {
    'rollup:options': (_, options) => {
      options.plugins = [
        options.plugins,
      ]
    },
  },
})
```
```
Cannot read properties of undefined (reading '_options')

// Locate this line of code
shebangPlugin2._options.preserve = false;
```
But `Vitejs` use code like it, [refer to Vite Repo](https://github.com/vitejs/vite/blob/50caf673f635f5a77f4cd72297c40fb4d77ead9b/packages/create-vite/build.config.ts#L25)
And the type also looks it should work:
```ts
type InputPluginOption = MaybePromise<Plugin | NullValue | false | InputPluginOption[]>
```

### Logs

```sh
ERROR  Error building D:/Code/test/unbuildErrTest: TypeError: Cannot read properties of undefined (reading '_options')                  23:25:55  


 ERROR  Cannot read properties of undefined (reading '_options')                                                                         23:25:55  

  at rollupBuild (/D:/Code/test/unbuildErrTest/node_modules/.pnpm/unbuild@2.0.0_typescript@5.4.2/node_modules/unbuild/dist/index.mjs:696:20)       
  at async _build (/D:/Code/test/unbuildErrTest/node_modules/.pnpm/unbuild@2.0.0_typescript@5.4.2/node_modules/unbuild/dist/index.mjs:1087:3)      
  at async build (/D:/Code/test/unbuildErrTest/node_modules/.pnpm/unbuild@2.0.0_typescript@5.4.2/node_modules/unbuild/dist/index.mjs:945:5)        
  at async Object.run (/D:/Code/test/unbuildErrTest/node_modules/.pnpm/unbuild@2.0.0_typescript@5.4.2/node_modules/unbuild/dist/cli.mjs:63:5)      
  at async runCommand (/D:/Code/test/unbuildErrTest/node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:316:16)
  at async runMain (/D:/Code/test/unbuildErrTest/node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:445:7)
```

**Comment by s3xysteak** [comment]
```ts
options.plugins = [
  ...options.plugins,
]
```
Can work but it will throw a type error, because `options.plugins` could be undefined. However, it seems it could not be undefined [refered](https://github.com/unjs/unbuild/blob/bfb4a348421826ad48072ca1cf70973e6caa5b1d/src/builder/rollup.ts#L319).
So I have to use it like:
```ts
options.plugins = [
  ...(options.plugins as any[] ),
  OtherPlugin()
]
```

**Comment by s3xysteak** [comment]
I read the code and found out the issues happened in [HERE](https://github.com/unjs/unbuild/blob/bfb4a348421826ad48072ca1cf70973e6caa5b1d/src/builder/rollup.ts#L194).
Raw code: 
```ts
// source code refer to above link.

rollupOptions.plugins = [
  rollupOptions.plugins,
  dts(ctx.options.rollup.dts),
  removeShebangPlugin(),
];
```
was transformed to 
```js
// dist/index.mjs

rollupOptions.plugins = rollupOptions.plugins || [];
// Panic here!
const shebangPlugin2 = rollupOptions.plugins.find(
  (p) => p && p.name === "unbuild-shebang"
);
shebangPlugin2._options.preserve = false;
rollupOptions.plugins.push(dts(ctx.options.rollup.dts));
```
`rollupOptions.plugins` can be nested, but it just use `find` here will cause issues. So `shebangPlugin2` will be undefined and `shebangPlugin2._options.preserve = false` try to get a attr of undefined will throw error.

**Comment by s3xysteak** [comment]
I just found out that the code in this main branch is not the source code for `v2.0.0` (I'm even confused about how it got compiled like this lol). I think maybe this issue won't exist in the new version, so I will test it.

**Comment by s3xysteak** [comment]
To fix the issue temporarily, change code here:  `(unbuild/dist/index.mjs:692)`

https://github.com/unjs/unbuild/blob/ff3eaa864b837060d93cffadb30e38e11db761cc/src/builder/rollup.ts#L181

to:
```ts
rollupOptions.plugins = rollupOptions.plugins?.flat(Infinity) || []
```

**Comment by s3xysteak** [comment]
Fixed in `v3.0.0-rc.1`.
